hugr-core/src/extension/op_def.rs

```diff
@@ -117,12 +117,9 @@ pub trait CustomLowerFunc: Send + Sync {
 /// Encode a signature as `PolyFuncTypeRV` but optionally allow validating type
 /// arguments via a custom binary. The binary cannot be serialized so will be
 /// lost over a serialization round-trip.
-#[derive(serde::Deserialize, serde::Serialize)]
 pub struct CustomValidator {
-    #[serde(flatten)]
     poly_func: PolyFuncTypeRV,
-    #[serde(skip)]
-    pub(crate) validate: Box<dyn ValidateTypeArgs>,
+    pub(crate) validate: Option<Box<dyn ValidateTypeArgs>>,
```

Comment:
If `None` here means "no custom validation required" then agreed, but in that case let's rename (or even inline) `struct CustomValidator`

Reply:
a rename or inline would be a breaking change, which certainly doesn't have to be
avoided but in this case I think just enough docstrings should do the trick?

Reply:
A breaking change just to people using the rust classes direct, but not to serialization (as this isn't serialized), right? I think that'd be OK, but see comment on `MissingValidateFunc` for an alternative "proposal"....

If this does affect serialization - I think the long-term plan here is to separate out a definition of serialization structure independent of this, in which case we can rename this without breaking serialization. But given this is no longer `#[serde]` I think this is not the case....